.github/workflows/custom_image_generation.yml

```diff
@@ -0,0 +1,159 @@
+
+    - name: Create svc acct file
+      run: 'echo "$GCP_DAISY_GCR_KEY" > "$GITHUB_WORKSPACE/jenkins/gce-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"'
+      shell: bash
+      env:
+        GCP_DAISY_GCR_KEY: ${{secrets.GCP_DAISY_GCR_KEY}}
+
+    - name: run script
+      run: |
+        gsutil ls $GCE_IMAGE_BUCKET || gsutil mb -p $GOOGLE_PROJECT -l $REGION $GCE_IMAGE_BUCKET
+        docker pull $DAISY_IMAGE
+        docker run -i --rm -v $GITHUB_WORKSPACE/jenkins/gce-custom-images:/gce-custom-images \
+        $DAISY_IMAGE \
+        -project $GOOGLE_PROJECT \
+        -zone $ZONE \
+        -oauth /gce-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME \
+        -gcs_path $GCE_IMAGE_BUCKET \
+        -default_timeout 45m \
+        -var:base_image $GCE_BASE_IMAGE \
+        -var:output_image "${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6}" \
+        -var:gce_images_dir /gce-custom-images \
+        -var:installation_script_name prepare_gce_image.sh \
+        /gce-custom-images/gce_image.wf.json
+        docker stop daisy || true
+        docker rm -f daisy || true
+        
+        # Daisy doesn't clean it up all so we remove the bucket manually
+        gsutil rm -r $GCE_IMAGE_BUCKET
+        
+        # Make the image public
+        gcloud beta compute images add-iam-policy-binding \
+        projects/${GOOGLE_PROJECT}/global/images/${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6} \
+        --member='allAuthenticatedUsers' \
+        --role='roles/compute.imageUser'
+
+  run-image-generate-script-dataproc:
+    runs-on: ubuntu-latest
+    # needs: inspect-changes
+    # if: needs.inspect-changes.outputs.has-dataproc-changes == 'true' && github.event.inputs.run-dataproc-override == 'true'
+    env:
+      DATAPROC_IMAGE_VERSION: 2.0.51-debian10
+      DATAPROC_IMAGE_VERSION_FORMATTED: 2-0-51-debian10
+      DATAPROC_IMAGE_BUCKET: gs://leo-dataproc-image-creation-logs
+      TEST_NAME: "${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6}"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: test step 2
+        run: echo "test 2"
+
+      - name: Create svc acct file
+        run: |
+          echo "$GCP_DAISY_GCR_KEY" > "$GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"
+          echo 'echoing test name'
+          echo "$TEST_NAME"
+          echo 'catting'
+          cat $GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME
+
+      - name: Init submodule
+        run: |
+          git submodule sync
+          git submodule update --init --recursive
+
+      - name: Run image generation script
+        run: |
+          docker run --rm -v $GITHUB_WORKSPACE/jenkins/dataproc-custom-images:/work:ro gcr.io/google.com/cloudsdktool/cloud-sdk:266.0.0 bash -c "cd /work/dataproc-custom-images \
+          && gcloud auth activate-service-account --key-file=/work/$GOOGLE_CREDENTIAL_FILE_NAME \
```

Comment:
I think this needs to be the same file path you setup in line 126. I don't think there is actually anything at this `/work/$GOOGLE_CREDENTIAL_FILE_NAME` path. Maybe a copy paste from Jenkins?

Reply:
I'm fairly sure line 139 is mounting the directory with the key into the docker image's `work` dir. I'll remove the `:ro` though, not sure what that is.

Reply:
Ah yeah I misread it, my bad. `:ro` means read-only volume